### DISCUSSION TITLE
How to handle issues when translator of kata is not available

### DISCUSSION THREAD

**Discussion by bupjae** [codewars/codewars.com#3095] [Q&A] [answered]
Context:

NASM translation of "Complete Binary Tree" ( https://www.codewars.com/kata/5c80b55e95eba7650dc671ea ) seems to broken.
The issue was reported as https://www.codewars.com/kata/5c80b55e95eba7650dc671ea/discuss/nasm#64b6865bf11bbe0017ce90a0 .
Original kata author said they don't know about NASM.
NASM translator of this kata doesn't seem to visit codewars.com anymore. (last seen Jul 2020)

What can be done when some tests are suspected as broken, and kata author/translator is no more available?

**Comment by hobovsky** [comment, marked as answer]
The original translator does not need to be active to fix the issue. Anyone can do this. The problem is that there might be not many users interested in fixing issues, and not many users able to fix issues with assembly translations. You can fix the issue too (challenging part might be that you need to solve the kata in the broken language first), use the "Fork" button, edit the code to fix the problem, publish the fork, announce the fix in kata discourse an on `#fixing` channel of Codewars Discord, and hopefully this will draw more attention to the case.

**Reply by bupjae** [reply]
> challenging part might be that you need to solve the kata in the broken language first

I don't think this is possible.
Even the stub solution that always returns zero-length result causes `System error: cannot allocate memory buffer.` error.

Then, should someone translate from start?

**Reply by hobovsky** [reply]
No, it is not necessary to translate from start. There is a group of users who can fix kata even if they havent solved them. I will try to get someone on it.

**Reply by hobovsky** [reply]
I attempted a fix, please verify if the kata is solvable now in NASM.

If there are still any errors, please raise an issue in the kata discourse.

**Reply by bupjae** [reply]
I confirmed that my code is accepted. Thanks for handling the issue.

**Comment by jessya020** [comment]
When a Codewars kata's tests (like this NASM-translated one) are broken and the original author/translator is unresponsive, here's what you can do:
First, double-check if the issue is indeed with the kata (not your solution) by verifying edge cases or cross-referencing with the original kata's logic. Then, post a detailed update in the kata's discussion thread (including specific test failures, expected vs. actual behavior) to flag the problem for moderators. Codewars moderators can review broken katas—if confirmed, they may temporarily disable the problematic language version, fix the tests themselves, or mark the kata for maintenance. You could also offer to contribute a fix (if you’re familiar with NASM) by submitting a suggestion via the discussion, which moderators can approve to resolve the issue.
